docs/src/content/docs/cookbook/dexes/stonfi.mdx

```diff
@@ -5,10 +5,362 @@ sidebar:
+
+Swaps have several common structures and messages. Don't forget to use them with the code examples below!
+
+```tact
+message(0x6664de2a) StonfiSwap {
+    otherTokenWallet: Address; // Address of the other Router token wallet
+    refundAddress: Address; // Address where refund will be sent if swap fails
+    excessesAddress: Address; // Address where TON excesses will be sent
+    deadline: Int as uint64; // Timestamp of execution deadline for this tx
+    additionalData: SwapAdditionalData; // Defined below
+}
+
+struct SwapAdditionalData {
+    minOut: Int as coins; // Minimum required amount of tokens to receive
+    receiver: Address; // Address where tokens will be sent after swap
+    fwdGas: Int as coins = 0; // Gas used to forward a message in transfer_notification after swap if custom_payload is present
+    customPayload: Cell? = null; // Payload sent in transfer_notification after swap
+    refundFwdGas: Int as coins = 0; // Gas used to forward a message in transfer_notification if swap fails if refund_payload is present
+    refundPayload: Cell? = null; // Payload sent in transfer_notification if swap fails
+    refFee: Int as uint16 = 10; // Referral fee. Max amount is 100 (1%)
+    referralAddress: Address? = null; // Referral address
+}
+
+message(0xf8a7ea5) JettonTransfer {
+    queryId: Int as uint64;
+    amount: Int as coins;
+    destination: Address;
+    responseDestination: Address?;
+    customPayload: Cell? = null;
+    forwardTonAmount: Int as coins;
+    forwardPayload: Cell?; // hack for now, equivalent to Slice as remaining with first bit set to 1
+}
+```
+
+Below are shown gas constants. All of them are taken from the sdk [STON.fi sdk](https://github.com/ston-fi/sdk). They are slightly differ from reality, but it is ok for now.
+
+```tact
+const GasSwapJettonToJetton: Int = ton("0.3");
+const GasSwapJettonToJettonFwd: Int = ton("0.24");
+
+const GasSwapJettonToTon: Int = ton("0.3");
+const GasSwapJettonToTonFwd: Int = ton("0.24");
+
+const GasSwapTonToJettonFwd: Int = ton("0.3");
+const GasSwapTonToJetton: Int = ton("0.01");
+```
+
+### Jetton to Jetton
+
+```tact
+const RouterAddress: Address = address("kQALh-JBBIKK7gr0o4AVf9JZnEsFndqO0qTCyT-D-yBsWk0v"); // CPI Router v2.1.0
+const RouterJettonWallet: Address = address("kQAtX3x2s-wMtYTz8CfmAyloHAB73vONzJM5S2idqXl-_5xK"); // Router Jetton Wallet Address
+
+message SwapJettonJetton {
+    myJettonWalletAddress: Address;
+}
+
+contract JettonToJetton {
+    receive() {}
+    receive(msg: SwapJettonJetton) {
+        // This example directly uses the `myJettonWalletAddress` provided in the received message.
+        // In real-world scenarios, it's more reliable to calculate this address on-chain or save it during initialization to ensure accuracy and trustworthiness.
+        let myJettonWalletAddress: Address = msg.myJettonWalletAddress;
+
+        let offerAmount: Int = 100000; // the quantity of jettons being swapped
+
+        let forwardPayload = StonfiSwap{
```

Comment:
`StonfiSwap` isn't present in this example block, which fails its compilation. See prior suggestions on explaining Structs/Messages on top of the page, then placing those repeated ones by the end of code blocks with examples — both to make them compile correctly and to simplify the initial top-down left-to-right read for the users